pyspectral/testing.py

```diff
@@ -0,0 +1,231 @@
+"""Utilities for users writing tests that interact with Pyspectral."""
+
+from __future__ import annotations
+
+import contextlib
+import os
+import shutil
+import tempfile
+from collections.abc import Iterator
+from pathlib import Path
+from typing import Callable
+from unittest import mock
+
+import numpy as np
+
+from pyspectral.rsr_reader import RSRDict
+from pyspectral.utils import ATMOSPHERES, RSR_DATA_VERSION, RSR_DATA_VERSION_FILENAME
+
+TMP_LUT_BASE_DIR = Path(tempfile.gettempdir()) / "pyspectral_fake_luts"
+
+
+@contextlib.contextmanager
+def mock_pyspectral_downloads(
+    tmp_path: Path = TMP_LUT_BASE_DIR,
+    rsr_data_version: str = RSR_DATA_VERSION,
+    extra_config_options: dict | None = None,
+    central_wavelengths: dict[str, float] | None = None,
+):
+    """Mock pyspectral's LUT downloads with fake realistic files."""
+    config_options = {}
+    # we want the tests to try to download files and create fake ones when that happens
+    config_options["download_from_internet"] = True
+
+    rsr_dir = tmp_path / "fake_rsr"
+    rsr_dir.mkdir(parents=True, exist_ok=True)
+    rayleigh_dir = tmp_path / "fake_rayleigh"
+    rayleigh_dir.mkdir(parents=True, exist_ok=True)
+    tb2rad_dir = tmp_path / "fake_tb2rad"
+    tb2rad_dir.mkdir(parents=True, exist_ok=True)
+    config_options["rsr_dir"] = str(rsr_dir)
+    config_options["rayleigh_dir"] = str(rayleigh_dir)
+    config_options["tb2rad_dir"] = str(tb2rad_dir)
+
+    if extra_config_options:
+        config_options.update(extra_config_options)
+
+    # setup rsr directory
+    rsr_version_path = rsr_dir / RSR_DATA_VERSION_FILENAME
+    with rsr_version_path.open("w") as rsr_version_file:
+        # defaults to active version, no downloading
+        rsr_version_file.write(rsr_data_version)
+
+    with (
+        mock.patch("pyspectral.utils._download_tarball_and_extract") as download_tarball,
+        override_config(config_options=config_options),
```

Comment:
should we pass the tmp_path instead of creating the temporary directory in override config?